### PR TITLE
Added instance domain to telemetry

### PR DESCRIPTION
**Pull request by hagen-danswer**
## Description
This pr adds a field called instance_domain to the anonymous_telemetry. This is only sent if EE features are enabled and telemetry is also enabled. the purpose is to help us track usage of our enterprise users per org (not per user)

### CODE REVIEW

File: .github/workflows/run-it.yml [hunk 1]
```diff
@@ -94,6 +94,7 @@ jobs:
           ENABLE_PAID_ENTERPRISE_EDITION_FEATURES=true \
           AUTH_TYPE=basic \
           REQUIRE_EMAIL_VERIFICATION=false \
```
Comment (hagen-danswer) [inline .github/workflows/run-it.yml:96]: this should disable telemetry spam from our integration tests

File: backend/danswer/configs/constants.py [hunk 2]
```diff
@@ -57,6 +57,7 @@
 KV_GEN_AI_KEY_CHECK_TIME = "genai_api_key_last_check_time"
 KV_SETTINGS_KEY = "danswer_settings"
 KV_CUSTOMER_UUID_KEY = "customer_uuid"
```
Comment (hagen-danswer) [inline backend/danswer/configs/constants.py:59]: the new value is stored in postgres as a kv so it stays consistent once generated

File: backend/danswer/utils/telemetry.py [hunk 3]
```diff
@@ -21,14 +28,43 @@ class RecordType(str, Enum):
     FAILURE = "failure"
 
 
-def get_or_generate_uuid() -> str:
+def _get_or_generate_uuid() -> str:
+    global _CACHED_UUID
```
Comment (hagen-danswer) [inline backend/danswer/utils/telemetry.py:32]: added caching for uuid generation

File: backend/danswer/utils/telemetry.py [hunk 4]
```diff
@@ -21,14 +28,43 @@ class RecordType(str, Enum):
...
+
     try:
-        return cast(str, kv_store.load(KV_CUSTOMER_UUID_KEY))
+        _CACHED_INSTANCE_DOMAIN = cast(str, kv_store.load(KV_INSTANCE_DOMAIN_KEY))
     except ConfigNotFoundError:
-        customer_id = str(uuid.uuid4())
-        kv_store.store(KV_CUSTOMER_UUID_KEY, customer_id, encrypt=True)
-        return customer_id
+        with Session(get_sqlalchemy_engine()) as db_session:
+            first_user = db_session.query(User).first()
```
Comment (hagen-danswer) [inline backend/danswer/utils/telemetry.py:60]: The logic to determine the "domain" of the danswer instance is to get the first user and use everything after the @ symbol of their email (E.g. danswer.ai)

File: backend/danswer/utils/telemetry.py [hunk 5]
```diff
@@ -41,16 +77,19 @@ def optional_telemetry(
...
+                customer_uuid = _get_or_generate_uuid()
                 payload = {
                     "data": data,
                     "record": record_type,
                     # If None then it's a flow that doesn't include a user
                     # For cases where the User itself is None, a string is provided instead
                     "user_id": user_id,
-                    "customer_uuid": get_or_generate_uuid(),
+                    "customer_uuid": customer_uuid,
                 }
```
Comment (hagen-danswer) [inline backend/danswer/utils/telemetry.py:88]: only run if EE is enabled (and Telemetry isnt disabled)

File: backend/danswer/utils/telemetry.py [hunk 6]
```diff
@@ -21,14 +28,43 @@ class RecordType(str, Enum):
     FAILURE = "failure"
 
 
-def get_or_generate_uuid() -> str:
+def _get_or_generate_uuid() -> str:
```
Comment (Weves) [inline backend/danswer/utils/telemetry.py]: can we generate this on startup?

File: backend/danswer/main.py [hunk 7]
```diff
@@ -324,6 +325,8 @@ async def lifespan(app: FastAPI) -> AsyncGenerator:
     # fill up Postgres connection pools
     await warm_up_connections()
 
+    get_or_generate_uuid()
```
Comment (Weves) [inline backend/danswer/main.py:329]: can we add a comment here about why we need this?

**Pull request merged**